### PR TITLE
feat(functions): add processFunctionLogs helper function

### PR DESCRIPTION
**Pull request by ace-n**
Move the `processFunctionLogs` function to the `php-tools` library.

Part of [this issue](https://github.com/GoogleCloudPlatform/php-docs-samples/issues/1271).

### CODE REVIEW

File: src/TestUtils/CloudFunctionDeploymentTrait.php [hunk 1]
```diff
@@ -133,4 +136,47 @@ public function getBaseUri()
...
+        // Define the log search criteria.
+        $logFullName = 'projects/' . $projectId . '/logs/cloudfunctions.googleapis.com%2Fcloud-functions';
+        $filter = sprintf(
+            'logName="%s" resource.labels.function_name="%s" timestamp>="%s"',
+            $logFullName,
+            self::$fn->getFunctionName(),
+            $startTime
+        );
+
+        echo "\nRetrieving logs [$filter]...\n";
```
Comment (bshaffer) [inline src/TestUtils/CloudFunctionDeploymentTrait.php:165]: nit: using "\n" here but PHP_EOL below. I prefer PHP_EOL but really just would like us to be consistent

File: src/TestUtils/GcloudWrapper/CloudFunction.php [hunk 2]
```diff
@@ -20,6 +20,7 @@
 use Symfony\Component\Process\Process;
 use Symfony\Component\Process\PhpExecutableFinder;
 use Symfony\Component\Process\Exception\ProcessFailedException;
+use PHPUnit\Framework\ExpectationFailedException;
```
Comment (bshaffer) [inline src/TestUtils/GcloudWrapper/CloudFunction.php:23]: Why was this added?  I don't see it referenced in this class.

File: src/TestUtils/CloudFunctionDeploymentTrait.php [hunk 3]
```diff
@@ -133,4 +136,47 @@ public function getBaseUri()
...
+
+    /**
+     * Retrieve and process logs for the defined function.
+     *
+     * @param string $startTime RFC3339 timestamp marking start of time range to retrieve.
+     * @param callable $process callback function to run on the logs.
+     */
+    private function processFunctionLogs(string $startTime, callable $process)
+    {
+        $projectId = getenv('GOOGLE_PROJECT_ID');
```
Comment (bshaffer) [inline src/TestUtils/CloudFunctionDeploymentTrait.php:148]: We should change this to `GOOGLE_CLOUD_PROJECT`

**Review by bshaffer — COMMENTED** [review]
a few small changes but otherwise looks good

**Review by grayside — APPROVED** [review]
minor issues discussed in meeting, nothing blocking.

File: src/TestUtils/GcloudWrapper/CloudFunction.php (hunk 2 above)
Comment (ace-n) [reply src/TestUtils/GcloudWrapper/CloudFunction.php]: Removed.

File: src/TestUtils/CloudFunctionDeploymentTrait.php (hunk 1 above)
Comment (ace-n) [reply src/TestUtils/CloudFunctionDeploymentTrait.php:165]: Fixed.

File: src/TestUtils/CloudFunctionDeploymentTrait.php (hunk 3 above)
Comment (ace-n) [reply src/TestUtils/CloudFunctionDeploymentTrait.php:148]: I assume we can address this later?

Comment (ace-n) [reply src/TestUtils/CloudFunctionDeploymentTrait.php]: Edit: done.
